extensions/amp-date-picker/1.0/component.js

```diff
@@ -0,0 +1,465 @@
+import {
+  addDays,
+  format as dateFnsFormat,
+  differenceInDays,
+  isAfter,
+  isSameDay,
+  isValid,
+  parse,
+} from 'date-fns';
+import {DayPicker} from 'react-day-picker';
+
+import {
+  closestAncestorElementBySelector,
+  scopedQuerySelector,
+} from '#core/dom/query';
+
+import * as Preact from '#preact';
+import {useCallback, useEffect, useMemo, useRef, useState} from '#preact';
+import {ContainWrapper} from '#preact/component';
+
+import './amp-date-picker.css';
+import {DatesList} from './dates-list';
+import {DayButton} from './day-button';
+import {AttributesContext} from './use-attributes';
+
+const DEFAULT_INPUT_SELECTOR = '#date';
+const DEFAULT_START_INPUT_SELECTOR = '#startdate';
+const DEFAULT_END_INPUT_SELECTOR = '#enddate';
+const ISO_8601 = 'yyyy-MM-dd';
+const FORM_INPUT_SELECTOR = 'form';
+// TODO: Check on this tag name
+const TAG = 'BentoDatePicker';
+const DEFAULT_ON_ERROR = (message) => {
+  throw new Error(message);
+};
+
+/** @enum {string} */
+const DatePickerMode = {
+  STATIC: 'static',
+  OVERLAY: 'overlay',
+};
+
+/** @enum {string} */
+const DatePickerType = {
+  SINGLE: 'single',
+  RANGE: 'range',
+};
+
+/** @enum {string} */
+const DateFieldType = {
+  DATE: 'input',
+  START_DATE: 'start-input',
+  END_DATE: 'end-input',
+};
+
+const DateFieldNameByType = {
+  [DateFieldType.DATE]: 'date',
+  [DateFieldType.START_DATE]: 'start-date',
+  [DateFieldType.END_DATE]: 'end-date',
+};
+
+/**
+ * @param {!BentoDatePicker.Props} props
+ * @return {PreactDef.Renderable}
+ */
+export function BentoDatePicker({
```

Comment:
This component is getting pretty massive, and I'm going to think about trying to divide up some of the responsibilities here.